code/modules/surgery/organs/organ_internal.dm

```diff
@@ -343,16 +343,47 @@
 	parent_organ = "head"
 	slot = "eyes"
 	var/list/eye_colour = list(0,0,0)
+	var/list/colourmatrix = null
+	var/list/colourblind_matrix = MATRIX_GREYSCALE //Special colourblindness parameters. By default, it's black-and-white.
+	var/colourblind_darkview = null
+	var/dependent_disabilities = null //Gets set by eye-dependent disabilities such as colourblindness so the eyes can transfer the disability during transplantation.
+	var/dark_view = 2 //Default dark_view for Humans.
+	var/weld_proof = null //If set, the eyes will not take damage during welding. eg. IPC optical sensors do not take damage when they weld things while all other eyes will.
 
 /obj/item/organ/internal/eyes/proc/update_colour()
 	dna.write_eyes_attributes(src)
 
-/obj/item/organ/internal/eyes/insert(mob/living/carbon/M, special = 0)
+/obj/item/organ/internal/eyes/proc/get_colourmatrix() //Returns a special colour matrix if the eyes are organic and the mob is colourblind, otherwise it uses the current one.
+	if(!robotic && owner.disabilities & COLOURBLIND)
+		return colourblind_matrix
+	else
+		return colourmatrix
+
+/obj/item/organ/internal/eyes/proc/get_dark_view() //Returns dark_view (if the eyes are organic) for see_invisible handling in species.dm to be autoprocessed by life().
+	if(!robotic && colourblind_darkview && owner.disabilities & COLOURBLIND) //Returns special darkview value if colourblind and it exists, otherwise reuse current.
+		return colourblind_darkview
+	else
+		return dark_view
+
+/obj/item/organ/internal/eyes/insert(mob/living/carbon/human/M, special = 0)
 	..()
 	if(istype(M) && eye_colour)
-		var/mob/living/carbon/human/H = M
-		// Apply our eye colour to the target.
-		H.update_body()
+		M.update_body() //Apply our eye colour to the target.
+
+	if(!(M.disabilities & COLOURBLIND) && (dependent_disabilities & COLOURBLIND)) //If the eyes are colourblind and we're not, carry over the gene.
+		dependent_disabilities &= ~COLOURBLIND
+		M.dna.SetSEState(COLOURBLINDBLOCK,1,1)
```

Comment:
Ah, almost missed this one - another deferred SE change that shouldn't be. The second `1` is the culprit.